File: src/algorithms/TwoSum/two_sum_problem.go
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, v := range nums {
		if _, ok := m[v]; ok && v**2 == target {
			return []int{i, m[v]}
		}
		m[v] = i
	}
	for k, v1 := range m {
		if v2, ok := m[target-k]; v1 != v2 && ok {
			return []int{v1, v2}
		}
	}
	return []int{}
}
	

File: src/algorithms/TwoSum/two_sum_problem.py

def two_sum(arr, S):
    sums = []
    for i in range(0, len(arr)):
        for j in range(i + 1, len(arr)):
            if (arr[i] + arr[j] == S):
                sums.append([arr[i], arr[j]])

    return sums


print(two_sum([3, 5, 2, -4, 8, 11], 7))
